clients/graphql/client.go

```diff
@@ -137,11 +140,23 @@ func fetchAlerts(
 		return nil, fmt.Errorf("returned error %v: %s", httpResp.Status, respBody)
 	}
 
-	// parse response
-	err = json.NewDecoder(httpResp.Body).Decode(resp)
+
+	// Check if the response is compressed with gzip
+	var respBodyReader *gzip.Reader
+	if strings.Contains(httpResp.Header.Get("Content-Encoding"), "gzip") {
+		respBodyReader, err = gzip.NewReader(httpResp.Body)
+		if err != nil {
+			return nil, err
+		}
+		defer respBodyReader.Close()
+	}
+
+	// Parse response
+	err = json.NewDecoder(respBodyReader).Decode(resp)
```

Comment:
If the response is not a gzip response (i.e. the header is missing), do we try to decode a nil reader here? We can avoid that if we do
```go
var respBodyReader io.Reader = httpResp.Body
```
at L145.